device_test.go

```diff
@@ -19,6 +19,11 @@ type deviceSuite struct {
 
 var _ = gc.Suite(&deviceSuite{})
 
+func (*deviceSuite) TestNilZone(c *gc.C) {
+	var empty device
+	c.Check(empty.Zone() == nil, jc.IsTrue)
```

Comment:
I tried that first, but it uses reflection so didn't fail before I fixed the problem.